subprojects/gradle/module-dependencies-graph/src/gradleTest/kotlin/com/avito/module/dependencies/DependenciesGraphTest.kt

```diff
@@ -0,0 +1,110 @@
+package com.avito.module.dependencies
+
+import com.avito.test.gradle.gradlew
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+
+internal class DependenciesGraphTest {
+
+    lateinit var projectDir: File
+
+    @BeforeEach
+    fun before(@TempDir dir: File) {
+        projectDir = dir
+        DependenciesGraphTestProject.generate(projectDir)
+    }
+
+    @Test
+    fun `print dependencies graph`() {
+        gradlew(projectDir, "printDependenciesGraph")
+            .assertThat()
+            .buildSuccessful()
+            .outputContains(
+                """
+                |> Task :printDependenciesGraph
+                |Root Node :CopyRootB
+                |:CopyRootB depends on :LeafG in ConfigurationType.AndroidTests
+                |:CopyRootB depends on :UniqueCopyB in ConfigurationType.AndroidTests
+                |:CopyRootB depends on :LeafC in ConfigurationType.Main
+                |:CopyRootB Node :LeafG
+                |:LeafG depends on :LeafE in ConfigurationType.Main
+                |:LeafG Node :LeafE
+                |:CopyRootB Node :UniqueCopyB
+                |:CopyRootB Node :LeafC
+                |:LeafC depends on :LeafD in ConfigurationType.Main
+                |:LeafC Node :LeafD
+                |:LeafD depends on :LeafE in ConfigurationType.Main
+                |:LeafD Node :LeafE
+                |Root Node :RootA
+                |:RootA depends on :LeafE in ConfigurationType.AndroidTests
+                |:RootA depends on :UniqueA in ConfigurationType.AndroidTests
+                |:RootA depends on :LeafF in ConfigurationType.Main
+                |:RootA depends on :LeafE in ConfigurationType.Main
+                |:RootA Node :LeafE
+                |:RootA Node :UniqueA
+                |:RootA Node :LeafF
+                |Root Node :RootB
+                |:RootB depends on :LeafG in ConfigurationType.AndroidTests
+                |:RootB depends on :UniqueB in ConfigurationType.AndroidTests
+                |:RootB depends on :LeafC in ConfigurationType.Main
+                |:RootB Node :LeafG
+                |:LeafG depends on :LeafE in ConfigurationType.Main
+                |:LeafG Node :LeafE
+                |:RootB Node :UniqueB
+                |:RootB Node :LeafC
+                |:LeafC depends on :LeafD in ConfigurationType.Main
+                |:LeafC Node :LeafD
+                |:LeafD depends on :LeafE in ConfigurationType.Main
+                |:LeafD Node :LeafE
+                |print ended""".trimMargin()
+            )
```

Comment:
Plain text output is hard to support in tests. It's fragile to changes.

What is the main subject of testing? I see it as a graph itself.
I'm thinking about an intermediate machine-readable format, use it to test logic.

Reply:
A good alternative option is to make it in a unit test by `ProjectBuilder`.

I've found an example with dependencies and plugins:

https://github.com/vanniktech/gradle-dependency-graph-generator-plugin/blob/master/src/test/java/com/vanniktech/dependency/graph/generator/DependencyGraphGeneratorTest.kt#L32

In this case, we don't need an output to test.

Reply:
I was thinking about that and I don't have a strong opinion yet.
I tried to compare our code with Gradle sources and I see there are only output tests.
Creating an intermediate format only for testing purpose is not a clear solution it has pros and cons

Reply:
Ok, what about `ProjectBuilder` example? Will it work?

Reply:
I will create a task